### PR TITLE
Kernel command line regex verification

### PR DESCRIPTION
**Pull request by k-naliuka**

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]
Thanks.

File: proto/attestation/verification.proto [hunk 1]
```diff
@@ -148,10 +148,8 @@ message KernelLayerData {
   // Measured digests of the setup data part of the kernel.
   RawDigest kernel_setup_data = 3;
 
-  // Measured digests of the command-line that was passed to the kernel
-  // during startup.
-  // TODO: b/325979696 - Validate the kernel command-line using a regex.
-  RawDigest kernel_cmd_line = 2 [deprecated = true];
+  // Command-line that was passed to the kernel during startup.
+  string kernel_cmd_line = 2;
```
Comment (tiziano88) [inline proto/attestation/verification.proto:152]: Should this use a different number?

File: proto/attestation/reference_value.proto [hunk 2]
```diff
@@ -89,6 +89,17 @@ message StringReferenceValue {
   repeated string values = 1;
 }
 
+message Regex {
+  string value = 1;
+}
+
+message RegexReferenceValue {
```
Comment (tiziano88) [inline proto/attestation/reference_value.proto:96]: I was thinking that we would have a StringReferenceValue, which may be either skip, or a literal string, or a regex. Not that the reference value itself would be about regex.

And the field name would just be `kernel_cmd_line` (instead of `kernel_cmd_line_regex`).

Anyways this can go in the bucket of future improvements.

File: oak_attestation_verification/tests/verifier_tests.rs [hunk 3]
```diff
@@ -474,3 +481,30 @@ fn verify_fails_with_empty_args() {
     assert!(r.is_err());
     assert!(p.status() == Status::GenericFailure);
 }
+
+#[test]
+fn verify_fails_with_command_line_reference_value_set() {
```
Comment (tiziano88) [inline oak_attestation_verification/tests/verifier_tests.rs:486]: Should we have a couple of test cases with a more realistic regex too? (can be in another PR)

**Review by tiziano88 — APPROVED** [review]

File: oak_attestation_verification/tests/verifier_tests.rs (hunk 3 above)
Comment (k-naliuka) [reply oak_attestation_verification/tests/verifier_tests.rs]: Added a more realistic test, with the caveat that it might need to be updated if the kernel command line for RK is :)

**Review by conradgrobler — APPROVED** [review]
